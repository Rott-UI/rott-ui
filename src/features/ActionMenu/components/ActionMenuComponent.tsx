// React Imports
import React, {FC} from 'react'

// Component Imports
import {Item, Pressable, Label, Button, List, ActionMenu} from '../..'
import {ActionMenuStyles} from '../styles'
import {ActionMenuProps, ActionModel} from '../models'

// Constant Imports

// Util and Lib Imports

/**
 * Action Menu
 *
 * @param title Başlık alanı
 * @param subTitle Alt başlık alanı
 * @param data ActionModel[] formatında veri listesi
 * @param visible ActionMenu komponentin bulunduğu yerdeki Show state i
 * @returns Action Menu Renderlanır
 *
 */
export const ActionMenuComponent: FC<ActionMenuProps> = ({
  title,
  subTitle,
  data,
  itemHeight,
  maxItem,
  separatorTotalHeight,
}) => {
  const itemCount = (data?.length ?? 0) > maxItem! ? maxItem! : data?.length ?? 0

  // Listede renderlanacak Item
  const renderItem = ({
    title: renderItemTitle,
    name: renderItemName,
    action: onAction,
  }: ActionModel) => (
    <Pressable
      testID={`action-menu-data-${renderItemName}-test-id`}
      height={itemHeight}
      justifyContentCenter
      alignItemsCenter
      onPress={() => onAction()}>
      <Label fontSize='lg' variant='grey-900' fontWeight={500}>
        {renderItemTitle}
      </Label>
    </Pressable>
  )

  return (
    <Item size='full' style={ActionMenuStyles().actionMenuContainer} alignItemsCenter>
      <Item
        width={342}
        backgroundColor={COLOURS.GREY100}
        borderTopStartRadius={!(title || subTitle) ? 12 : 0}
        borderTopEndRadius={!(title || subTitle) ? 12 : 0}
        borderBottomStartRadius={12}
        borderBottomEndRadius={12}
        marginBottom={16}>
        <List
          testID='action-menu-test-id'
          style={{minHeight: 2}}
          width={342}
          height={itemHeight! * itemCount + separatorTotalHeight!}
          data={data}
          renderItem={({item}) => renderItem(item)}
          horizontal={false}
          separatorVariant='neutral-alpha-200'
          renderSeparator
          headerSeparator
          footerSeparator
          estimatedItemSize={display.normalize(itemHeight!, 'height')}
          showsVerticalScrollIndicator={false}
          scrollEnabled={(data?.length ?? 0) > maxItem!}
        />
      </Item>

      <Button
        variant='primary'
        onPress={() => ActionMenu.hideActionMenu()}
        fontSize='xl'
        size='full'
        testID='action-menu-cancel-test-id'
        marginBottom={21}>
        {translator('COMMON.CANCEL')}
      </Button>
    </Item>
  )
}
