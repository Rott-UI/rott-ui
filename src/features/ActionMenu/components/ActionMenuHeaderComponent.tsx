// React Imports
import React, {FC} from 'react'

// Util and Lib Imports
import {Item, Label} from '../..'
import {themeConfig} from '../../../providers'

interface ActionMenuHeaderProps {
  title?: Nullable<string>
  subTitle?: Nullable<string>
}

export const ActionMenuHeaderComponent: FC<ActionMenuHeaderProps> = ({title, subTitle}) => {
  return (
    <Item size='full' alignItemsCenter>
      <Item testID='action-menu-header-test-id' width={342}>
        {(title || subTitle) && (
          <Item
            justifyContentCenter
            alignItemsCenter
            size='full'
            backgroundColor={themeConfig.colors['grey-100']}
            borderTopStartRadius={12}
            borderTopEndRadius={12}
            height={54}>
            {title && (
              <Label
                fontSize='xl'
                variant='grey-900'
                fontWeight={700}
                marginBottom={subTitle ? 4 : undefined}>
                {title}
              </Label>
            )}
            {subTitle && (
              <Label fontSize='sm' variant='grey-900' fontWeight={500}>
                {subTitle}
              </Label>
            )}
          </Item>
        )}
      </Item>
    </Item>
  )
}
