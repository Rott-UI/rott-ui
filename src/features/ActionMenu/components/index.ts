export * from './ActionMenuComponent'
export * from './ActionMenuHeaderComponent'
