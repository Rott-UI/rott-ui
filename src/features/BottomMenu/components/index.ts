export * from './BottomMenu'
export * from './BottomMenuItem'
