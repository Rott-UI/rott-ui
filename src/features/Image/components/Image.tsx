// React Imports
import {FC} from 'react'

// React Native Imports
import {
  ImageProps as RNImageProps,
  Image as RNImage,
  StyleSheet,
  ImageSourcePropType,
} from 'react-native'

import {CommonUiProps} from '../../models'

// Style Imports
import {ImageStyles} from '../styles'
import {ImageTypes} from '../models'
import {Images} from '../../../constants'
import {IconTypes} from '../../Icon'

interface ImageProps
  extends Omit<RNImageProps, 'name' | 'source'>,
    Omit<CommonUiProps, 'name' | 'borderRadius' | 'width' | 'height'> {
  name?: ImageTypes | IconTypes // TODO: KONTROL
  source?: ImageSourcePropType
}
export const Image: FC<ImageProps> = ({variant, size, style, source, name, ...props}) => {
  const imageSource = Object.entries(Images).find(
    (image) => image[0]?.toLowerCase() === name?.toLowerCase()
  )

  return (
    <RNImage
      style={StyleSheet.flatten([ImageStyles({variant, size, ...props}).defaultImageStyle, style])}
      source={source ?? Images[imageSource?.[0] as never]}
      {...props}
    />
  )
}
