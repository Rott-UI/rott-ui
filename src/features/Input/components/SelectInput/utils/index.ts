export * from './listHeightNormalizer'
export * from './modalHeightPercentageNormalizer'
