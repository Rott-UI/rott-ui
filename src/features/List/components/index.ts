export * from './List'
export * from './FavoriteSwipeComponent'
