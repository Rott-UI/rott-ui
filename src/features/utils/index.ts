export * from './commonUiTestExtension'
export * from './commonUiStyleProperties'
