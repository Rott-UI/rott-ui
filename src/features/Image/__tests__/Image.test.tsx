// Component Imports
import {Image} from '../components'

// Util and Lib Imports
import {render} from '../../../utils'

describe('Image -> Custom Component', () => {
  const testId = {
    imageTestId: 'image-test-id',
  }
  const testIconName = 'ARROW_LEFT'

  it('image elementi snapshot ile eşleşmeli', () => {
    const {imageTestId} = testId
    const renderedIcon = render(
      <Image testID={imageTestId} name={testIconName} width={50} height={50} />
    )

    expect(renderedIcon).toMatchSnapshot()
  })

  it('image elementi renderlandığında gerçekten ekranda olmalı', () => {
    const {imageTestId} = testId
    const {getByTestId} = render(<Image testID={imageTestId} name={testIconName} width={50} />)

    const imageElement = getByTestId(imageTestId)

    expect(imageElement).toBeOnTheScreen()
  })

  it('image elementi verilen width ile renderlanmalı', () => {
    const {imageTestId} = testId
    const {getByTestId} = render(<Image testID={imageTestId} name={testIconName} width={50} />)

    const imageElement = getByTestId(imageTestId)

    expect(imageElement).toHaveProp('width', 50)
  })

  it('image elementi verilen height ile renderlanmalı', () => {
    const {imageTestId} = testId
    const {getByTestId} = render(<Image testID={imageTestId} name={testIconName} height={50} />)

    const imageElement = getByTestId(imageTestId)

    expect(imageElement).toHaveProp('height', 50)
  })
})
