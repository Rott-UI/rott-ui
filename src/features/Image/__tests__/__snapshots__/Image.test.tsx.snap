// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image -> Custom Component image elementi snapshot ile eşleşmeli 1`] = `
[
  <Image
    height={50}
    style={
      {
        "backgroundColor": undefined,
        "borderRadius": 50,
        "bottom": undefined,
        "flexGrow": undefined,
        "gap": undefined,
        "height": 96,
        "left": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "maxHeight": 96,
        "maxWidth": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "overflow": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
        "position": undefined,
        "resizeMode": "contain",
        "right": undefined,
        "top": undefined,
        "width": 96,
        "zIndex": undefined,
      }
    }
    testID="image-test-id"
    width={50}
  />,
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      [
        {
          "alignSelf": "center",
          "elevation": 999999,
          "flex": 0,
          "maxWidth": "100%",
          "position": "absolute",
          "zIndex": 999999,
        },
        {
          "flexDirection": "column-reverse",
          "justifyContent": "flex-start",
          "top": 30,
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        {
          "bottom": 0,
        }
      }
    >
      <RCTSafeAreaView />
    </View>
  </View>,
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      [
        {
          "alignSelf": "center",
          "elevation": 999999,
          "flex": 0,
          "maxWidth": "100%",
          "position": "absolute",
          "zIndex": 999999,
        },
        {
          "bottom": 40,
          "flexDirection": "column",
          "justifyContent": "flex-end",
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        {
          "bottom": 0,
        }
      }
    >
      <RCTSafeAreaView />
    </View>
  </View>,
]
`;
